tests/llm/test_example_selector.py

```diff
@@ -0,0 +1,218 @@
+            [{"input": '{"operand_1":0.0,"operand_2":8.0}', "output": ResponseModel(sum=8.0, prod=0.0)}]
+        )
+
+        ground_truth = [
+            Example(input='{"operand_1":0.0,"operand_2":8.0}', output=ResponseModel(sum=8.0, prod=0.0)),
+            Example(input=RequestModel(operand_1=-1.0, operand_2=-2.0), output=ResponseModel(sum=-3.0, prod=2.0)),
+        ]
+
+        await selector.aadd_example(
+            {"input": RequestModel(operand_1=-1.0, operand_2=-2.0), "output": ResponseModel(sum=-3.0, prod=2.0)}
+        )
+
+        assert selector.root == ground_truth
+
+    async def test_aselect_examples(self, static_selector_fixture):
+
+        selector = static_selector_fixture(
+            [
+                {"input": '{"operand_1":0.0,"operand_2":8.0}', "output": ResponseModel(sum=8.0, prod=0.0)},
+                {"input": RequestModel(operand_1=-1.0, operand_2=-2.0), "output": ResponseModel(sum=-3.0, prod=2.0)},
+            ]
+        )
+
+        ground_truth = [
+            {"input": '{"operand_1":0.0,"operand_2":8.0}', "output": '{"sum":8.0,"prod":0.0}'},
+            {"input": '{"operand_1":-1.0,"operand_2":-2.0}', "output": '{"sum":-3.0,"prod":2.0}'},
+        ]
+        result = await selector.aselect_examples(input_variables={})
+
+        assert result == ground_truth
+
+
+class TestToLangchainContext:
+
+    async def test_empty_selector(self, static_selector_fixture):
+
+        selector = static_selector_fixture([])
+
+        messages = await to_langchain_context(example_selector=selector, input_variables={})
+        assert messages == []
+
+    async def test_selector_with_content(self, static_selector_fixture):
+
+        selector = static_selector_fixture([{"input": "7, 6", "output": "13"}, {"input": "8, -9", "output": "-1"}])
+
+        ground_truth = [
+            HumanMessage(content="7, 6", additional_kwargs={}, response_metadata={}),
+            AIMessage(content="13", additional_kwargs={}, response_metadata={}),
+            HumanMessage(content="8, -9", additional_kwargs={}, response_metadata={}),
+            AIMessage(content="-1", additional_kwargs={}, response_metadata={}),
+        ]
+
+        messages = await to_langchain_context(example_selector=selector, input_variables={})
+        assert messages == ground_truth
+
+    async def test_langchain_selector(self):
+
+        example_prompt = PromptTemplate(input_variables=["input", "output"], template="Input: {input}\nOutput: {output}")
+
+        selector = LengthBasedExampleSelector(
+            examples =  [
+                {"input": '{"operand_1":3.0,"operand_2":4.0}', "output": '{"sum":7.0,"prod":12.0}'},
+                {"input": '{"operand_1":0.0,"operand_2":8.0}', "output": '{"sum":8.0,"prod":0.0}'},
+                {"input": '{"operand_1":5.0,"operand_2":6.0}', "output": '{"sum":11.0,"prod":30.0}'},
+            ],
+            example_prompt = example_prompt,
+            max_length= 11
```

Comment:
Those are strange spaces.
It's recommended to not use any spaces around `=` when there are no type annotations:
https://peps.python.org/pep-0008/#other-recommendations